src/CodeGenerator/Command/GenerateCommand.php

```diff
@@ -103,7 +103,13 @@ protected function execute(InputInterface $input, OutputInterface $output)
             $baseNamespace = $manifest['services'][$serviceName]['namespace'] ?? \sprintf('AsyncAws\\%s', $serviceName);
             $resultNamespace = $baseNamespace . '\\Result';
 
-            $this->generator->client()->generate($definition, $serviceName, $baseNamespace);
+
+            $clientGenerator = $this->generator->client();
+            $operationGenerator = $this->generator->operation();
+            $resultGenerator = $this->generator->result();
+            $waiterGenerator = $this->generator->waiter();
```

Comment:
I see that you do this to avoid getting new instances of these objects. What if you did this "caching" in `ApiGenerator` instead?